src/lib/services/public-signup-token-service.ts

```diff
@@ -0,0 +1,109 @@
+import crypto from 'crypto';
+import { Logger } from '../logger';
+import { IUnleashConfig, IUnleashStores } from '../types';
+import { minutesToMilliseconds } from 'date-fns';
+import { IPublicSignupTokenStore } from '../types/stores/public-signup-token-store';
+import { PublicSignupTokenSchema } from '../openapi/spec/public-signup-token-schema';
+import { IRoleStore } from '../types/stores/role-store';
+import { IPublicSignupTokenCreate } from '../types/models/public-signup-token';
+import { PublicSignupTokenCreateSchema } from '../openapi/spec/public-signup-token-create-schema';
+import { RoleName } from '../types/model';
+
+export class PublicSignupTokenService {
+    private store: IPublicSignupTokenStore;
+
+    private roleStore: IRoleStore;
+
+    private logger: Logger;
+
+    private timer: NodeJS.Timeout;
+
+    private activeTokens: PublicSignupTokenSchema[] = [];
+
+    constructor(
+        {
+            publicSignupTokenStore,
+            roleStore,
+        }: Pick<IUnleashStores, 'publicSignupTokenStore' | 'roleStore'>,
+        config: Pick<IUnleashConfig, 'getLogger' | 'authentication'>,
+    ) {
+        this.store = publicSignupTokenStore;
+        this.roleStore = roleStore;
+        this.logger = config.getLogger(
+            '/services/public-signup-token-service.ts',
+        );
+        this.fetchActiveTokens();
+        this.timer = setInterval(
+            () => this.fetchActiveTokens(),
+            minutesToMilliseconds(1),
+        ).unref();
+    }
+
+    async fetchActiveTokens(): Promise<void> {
+        try {
```

Comment:
Is this try catch needed? We try catch in the base controller which wraps the services. Seems like this could be omitted.

Reply:
Removed. Thanks!